server/audit_math/suite.py

```diff
@@ -1,718 +0,0 @@
-
-    for rnd in bp_stratum.sample:
-        hyp_sample[rnd] = bp_stratum.sample[rnd]
-
-    hyp_no_cvr_stratum = BallotPollingStratum(
-        bp_stratum.num_ballots, bp_stratum.vote_totals, hyp_sample, hyp_sample_size
-    )
-
-    return maximize_fisher_combined_pvalue(
-        alpha, contest, hyp_no_cvr_stratum, hyp_cvr_stratum, winner, loser
-    )
-
-
-def get_sample_size(
-    risk_limit: int,
-    contest: Contest,
-    bp_stratum: BallotPollingStratum,
-    cvr_stratum: BallotComparisonStratum,
-) -> HybridPair:
-    """
-    Estimate the initial sample sizes for the audit.
-
-    Inputs:
-        risk_limit      - the risk limit (as an integer percentage) for this audit
-        contest         - the overall contest information
-        bp_stratum: a stratum object containing the ballot polling stratum
-        cvr_stratum: a stratum object containing the ballot comparison stratum
-    Outputs:
-        sample_sizes    - A Tuple of (cvr_strata_size, no_cvr_strata_size).
-    """
-
-    alpha = float(risk_limit) / 100
-
-    for winner, loser in product(contest.winners, contest.losers):
-        n_ratio = cvr_stratum.num_ballots / (
-            cvr_stratum.num_ballots + bp_stratum.num_ballots
-        )
-        num_sampled = max(
-            MIN_SAMPLE_SIZE, cvr_stratum.sample_size + bp_stratum.sample_size
-        )
-
-        ballots_to_sample = num_sampled
-
-        expected_pvalue = 1.0
-
-        # this allows us to exactly match CORLA18's estimate_n and estimate_escalation_n
-        coefficient = 1.1
-        if bp_stratum.sample_size == 0 and cvr_stratum.sample_size == 0:
-            coefficient = 2.0
-
-        # step 1: linear search, increasing n by a factor of 1.1 each time
-        while (expected_pvalue > alpha) or (expected_pvalue is np.nan):
-            ballots_to_sample = int(coefficient * ballots_to_sample)
-            if ballots_to_sample > contest.ballots:
-                cvr_ballots_to_sample = math.ceil(n_ratio * contest.ballots)
-                bp_ballots_to_sample = int(contest.ballots - cvr_ballots_to_sample)
-                return HybridPair(
-                    cvr=cvr_ballots_to_sample, non_cvr=bp_ballots_to_sample
-                )
-
-            expected_pvalue = try_n(
-                ballots_to_sample,
-                alpha,
-                contest,
-                winner,
-                loser,
-                bp_stratum,
-                cvr_stratum,
-                n_ratio,
-            )
-
-        # step 2: bisection between n/1.1 and n
-        low_n = ballots_to_sample / coefficient
-        high_n = ballots_to_sample
-        mid_pvalue = 1.0
-        # TODO: do we need this tolerance?
-        risk_limit_tol = 0.8
-        # while (mid_pvalue > alpha) or (expected_pvalue is np.nan):
-        while (
-            (mid_pvalue > alpha)
-            or (mid_pvalue < risk_limit_tol * alpha)
-            or (expected_pvalue is np.nan)
```

Comment:
is this supposed to be `expected_pvalue` or `mid_pvalue`?

Reply:
`expected_pvalue`: https://github.com/pbstark/CORLA18/blob/master/code/suite_tools.py#L421

Since the first loop breaks if it's `nan`, it might be `nan` in the first iteration and cause the other two checks to fail.

Reply:
i think the first loop actually _continues_ if `expected_pvalue` is `nan`

when is a pvalue `nan`? that might help us make sense of this

Reply:
Oh shoot you're right. Looking at the code again, I'm not sure we can actually encounter a `nan`? Like it should only happen if a p-value is negative, which can't actually happen. So we can probably take these checks out.